test/widgets/dialog_test.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/material.dart';
+import 'package:zulip/widgets/dialog.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../model/binding.dart';
+import 'dialog_checks.dart';
+import 'test_app.dart';
+
+void main() {
```

Comment:
Cool, these tests look broadly right.

Let's have them check one other important aspect: each of the buttons in the dialog does the intended thing if you tap it.